src/nats.cr

```diff
@@ -360,12 +358,12 @@ module NATS
     #   response.status = :service_unavailable
     # end
     # ```
-    def request(subject : String, message : Data = "", timeout : Time::Span = 2.seconds, headers : Headers? = nil) : Message?
+    def request(subject : String, data : Payload = "", timeout : Time::Span = 2.seconds, headers : Headers? = nil) : Message?
```

Comment:
I kept the internal alias name as `Payload` because `data : Data` in the function signature seems a little ambiguous.  The Go client uses "payload" several times when needed for clarity.